### ISSUE TITLE
Fix looping for short HTML5 GIFs

### ISSUE THREAD

**Issue by akaleeroy** [ubershmekel/redditp#56]
**Current behavior:**
Short GIFs in video format play once and immediately skip to next slide, disregarding `rp.settings.timeToNextSlide`

**Expected behavior:**
GIFs should loop for as long as the `rp.settings.timeToNextSlide` is. Regardless of whether they are an actual image file with a .gif extension or a .webm / .mp4  video.

See [this fix](https://github.com/DanTheBlue/redditp/commit/27be17b3d32620c2212eb77ed9668569f8824816#diff-2f86cb96a0233e7cb3b6f03ad573be0bR504) in a fork

**Comment by ubershmekel** [comment]
I can imagine we might want a few different settings:

* Max loop count (now is effectively 1)
* Max seconds playing (currently only works for images, not video gifs)
* Minimum seconds (loop at least until this time, ignore max loop count)

I'm not sure if exposing all these is very user friendly. But I'd be interested to hear suggestions. If we find an elegant solution, we could implement it. I agree that the default `Minimum seconds` should be 1-2 seconds to represent short gifs well.

**Comment by akaleeroy** [comment]
Cinemagraphs can be only a few frames long, but you're meant to watch them for ages like screensavers. If you let them play 1 loop only, from start to finish, it's a flash. 1-2 seconds is also too short. 
The best way is to play for as long as it says in the `#timeToNextSlide` box! User sets that and there are no surprises.

Advancing to next slide in the middle of the duration (instead of cleanly at the end) is also not necessary. These videos are meant to be GIFs, which don't have that option anyway.

I worked in [#f08897b](https://github.com/DanTheBlue/redditp/commit/27be17b3d32620c2212eb77ed9668569f8824816#diff-2f86cb96a0233e7cb3b6f03ad573be0bR504) from [@DanTheBlue](https://github.com/DanTheBlue/) and it's a good behavior. Slide advances after the time you specified. Used [Flip the Script](https://chrome.google.com/webstore/detail/flip-the-script/bmkpjhjogcjpioeffjmdgclmeiiofcjp?hl=en "Flip the Script - Chrome Web Store") to substitute redditp.com's `script.js` with a local copy.

**Comment by Hrxn** [comment]
For stuff like cinemagraphs etc. I also think that using the delay specified by the user makes the most sense. +1

**Comment by ubershmekel** [comment]
I would like to support the option of "go to the next gif after this one is over" as well.

**Comment by Hrxn** [comment]
Depending on how quickly/reliably the 'runtime' of a GIF can be determined:

If runtime-gif < user-delay
    time-to-next = user-delay
else
    time-to-next = runtime-gif
end

**Comment by ubershmekel** [comment]
@Hrxn You're suggesting the current delay would mean `wait at least this long or longer until it finishes`. I think some people would like a `wait at most this long, cutting off a gif that is too long` setting. I'd like a nice way to represent this in the UI which already feels overloaded. Maybe the right thing to do is to only give one of these options. I'm not sure.

**Comment by Hrxn** [comment]
Yes, that is what I'm saying. That is the current behaviour for all 'video as animation', e.g. `.gifv` from Imgur or Gfycat, at least for me. 

This is the most sensible variant, in my opinion. I don't see the point in cutting off an item. But if someone requests this, I think a small checkbox with `enforce` or something next to the time entry field would be the most intuitive solution.

**Comment by hppavilion1** [comment]
@Hrxn In the event that the gif's length isn't an integer divisor of the time_to_next, should the gif be allowed to loop past the specified maximum so it can complete a cycle?

**Comment by hppavilion1** [comment]
Perhaps a simple "Go to next immediately when GIF completes" (though with better phrasing, I'm sure) checkbox would be good enough?

**Comment by ubershmekel** [comment]
A pull request fixing redditp to the @Hrxn suggested behavior would be accepted.

Any clever checkbox or menu options would have to be designed in an extensible and subtle way so would likely require multiple iterations.

**Comment by nofapexpert** [comment]
I would like to raise this issue again because it's still not completely fixed. Apparently as of now, GIFs will be looped but MP4 videos are only played once.

I agree with @akaleeroy , the default behavior should be to play any video for EXACTLY the time specified by the user. If it is too short, loop it. If it is too long, cut it. An option to "play videos once" would cover both cases, short GIFs as well as long videos, so they will be played exactly one regardless of their length. :)
